### PR TITLE
Feature/simple interest calculator

### PR DESCRIPTION
**Pull request by Mustafa-Mangal**
---
name: Solution Review 
about: A template PR for code review with a checklist
---

### Challenge Solved:
This PR implements the solution for the simple interest calculator coding challenge. The task is to calculate simple interest based on principal, interest rate, and time in years using the formula:

## Simple Interest Formula

The formula for calculating simple interest is:
**I** = _`P × R × T / 100`_
Where:
- \( I \) is the interest
- \( P \) is the principal amount
- \( R \) is the rate of interest per period
- \( T \) is the time the money is invested or borrowed for, in periods

### Changes Implemented:
- Implemented the simple interest calculation logic in `simple_interest_calculator.py`.
- Added unit tests to validate the calculator’s functionality in `test_simple_interest_calculator.py`.

#### Files

- [ ] The file name (`simple_interest_calculator.py`) describes the function's behavior.
- [ ] The test file is named `test_simple_interest_calculator.py` and matches the function file.
- [ ] Both files contain module-level docstrings.

#### Unit Tests

- [ ] The test class is named `TestSimpleInterestCalculator` (PascalCase).
- [ ] The test class includes a docstring.
- [ ] Each test case has a descriptive name and clear docstrings.
- [ ] All tests pass successfully.
- [ ] Tests cover defensive assertions and boundary cases.

#### Function Docstring

- [ ] The function's behavior is well-described in the docstring.
- [ ] Function arguments and return values are clearly documented, including types and assumptions.
- [ ] Proper defensive assertions are included, and they are well-documented.

#### Do's

- [ ] Variable names such as `principal`, `rate`, and `time` are clear and descriptive.
- [ ] The code is organized, and the logic is straightforward, ensuring simplicity.

#### Don'ts

- [ ] No unnecessary comments that explain the implementation; the code is self-explanatory.

### Implementation

- [ ] The code passes formatting and linting checks.
- [ ] No significant Pylint errors.
- [ ] Variables follow snake_case conventions.
- [ ] The code implements the solution in a simple and clean manner.

---

### Review Checklist for Merge:

- [ ] Code passes all tests
- [ ] Linting and formatting checks passed
- [ ] Code reviewed by team member(s)
- [ ] PR linked to corresponding issue on project board

### CODE REVIEW

**Comment by Elocodes** [comment]
Regarding the checks that are not passing, I suggest you format your file with either the ruff or black formatter. Have you tried this?

Secondly, seems like you created extra folders (src and tests) for your challenge. This may cause confusion for the team. why not add your challenge to the body of the existing solutions folder and place your test case into the existing test folder?

File: README.md
Comment (Elocodes) [inline README.md:1]: The general group repository Readme.md should explain the entire team project not a specific challenge as you have done here. consider moving this to the notes folder

File: src/simple_interest_calculator.py
Comment (Elocodes) [inline src/simple_interest_calculator.py:1]: creating an added folder "src" for your specific challenge would disrupt team's project folder structure. consider moving the simple_interest_calculator.py file into the solutions folder and removing the "src" folder you created

File: tests/test_simple_interest_calculator.py
Comment (Elocodes) [inline tests/test_simple_interest_calculator.py:1]: To maintain same project folder structure for everyone, consider moving this file "test_simple_interest_calculator.py" into the already existing "tests" subfolder found in the solutions folder

Comment (Elocodes) [inline tests/test_simple_interest_calculator.py:1]: Your unit tests are missing docstrings. Please note that comments in python are not the same as docstrings

**Review by Elocodes — CHANGES_REQUESTED** [review]

File: README.md
Comment (Mustafa-Mangal) [reply README.md:1]: Hi
I recently reorganized my project files for better clarity. The source files were moved from the Src folder to the Solutions folder, while the test files were relocated to the Tests folder. Additionally, the README file was moved to the Tests folder and renamed to reflect the task name.

File: src/simple_interest_calculator.py
Comment (Mustafa-Mangal) [reply src/simple_interest_calculator.py:1]: Point it and moved the files to the designated folders

File: tests/test_simple_interest_calculator.py
Comment (Mustafa-Mangal) [reply tests/test_simple_interest_calculator.py:1]: Agree and done

Comment (Mustafa-Mangal) [reply tests/test_simple_interest_calculator.py:1]: Resolved

**Review by Mustafa-Mangal — COMMENTED** [review]
Resolved the suggested changes

File: collaboration/README.md [hunk 1]
```diff
@@ -1,102 +1,61 @@
...
+- Address the project, not the person - creates a safe environment for achieving
+  common project goal
 
-- Address the project not the person - creates a safe environment for achieving the
-common project goal
-
-</details>
-
-<details>
-<summary> What we do if that trust is broken  </summary>
```
Comment (Mustafa-Mangal) [inline collaboration/README.md:18]: The formatting was updated to replace `<details>` and `<summary>` HTML tags with Markdown headings and lists. This ensures compliance with Markdown linting rules and improves compatibility across different Markdown parsers.

File: collaboration/README.md [hunk 2]
```diff
@@ -1,102 +1,61 @@
...
-
-- Aside contentious issues as laid out above, team should be able to discuss things
-normally and agree on what works for everyone
-
-</details>
-
-<details>
-<summary> How the team approaches meeting times and agendas
-
-  </summary>
```
Comment (Mustafa-Mangal) [inline collaboration/README.md:54]: The formatting was updated to replace `<details>` and `<summary>` HTML tags with Markdown headings and lists. This ensures compliance with Markdown linting rules and improves compatibility across different Markdown parsers.

File: collaboration/README.md [hunk 3]
```diff
@@ -1,102 +1,61 @@
...
+## How and where the team communicates about different aspects of the project?
 
-- Slack and Whatsapp are the two major means of communication for the team
+- Slack and Whatsapp are the two major means of communication for the team.
 
-</details>
-
-<details>
-<summary> Team's Respect Codes
-  </summary>
```
Comment (Mustafa-Mangal) [inline collaboration/README.md:70]: # Explanation for Formatting Changes

The formatting was updated to replace `<details>` and `<summary>` HTML tags with Markdown headings and lists. This ensures compliance with Markdown linting rules and improves compatibility across different Markdown parsers.

**Review by Mustafa-Mangal — COMMENTED** [review]
The formatting was updated to replace `<details>` and `<summary>` HTML tags with Markdown headings and lists. This ensures compliance with Markdown linting rules and improves compatibility across different Markdown parsers.

**Review by Elocodes — APPROVED** [review]
